runtime/near-evm-runner/src/builtins.rs

```diff
@@ -0,0 +1,576 @@
+use byteorder::{BigEndian, ByteOrder, LittleEndian, ReadBytesExt};
+use ethereum_types::{Address, H256, U256};
+use num_bigint::BigUint;
+use num_traits::{One, Zero};
+
+use parity_bytes::BytesRef;
+use ripemd160::Digest;
+use std::{
+    cmp::{max, min},
+    io::{self, Cursor, Read},
+    mem::size_of,
+};
+use vm::{MessageCallResult, ReturnData};
+
+pub static COUNT: u64 = 9;
+
+pub fn is_precompile(addr: &Address) -> bool {
+    *addr <= Address::from_low_u64_be(COUNT)
+}
+
+pub fn precompile(id: u64) -> Box<dyn Impl> {
+    match id {
+        1 => Box::new(EcRecover) as Box<dyn Impl>,
+        2 => Box::new(Sha256) as Box<dyn Impl>,
+        3 => Box::new(Ripemd160) as Box<dyn Impl>,
+        4 => Box::new(Identity) as Box<dyn Impl>,
+        5 => Box::new(ModexpImpl) as Box<dyn Impl>,
+        6 => Box::new(Bn128AddImpl) as Box<dyn Impl>,
+        7 => Box::new(Bn128MulImpl) as Box<dyn Impl>,
+        8 => Box::new(Bn128PairingImpl) as Box<dyn Impl>,
+        9 => Box::new(Blake2FImpl) as Box<dyn Impl>,
+        _ => panic!("Invalid builtin ID: {}", id),
```

Comment:
Might be better to use enum as number: https://doc.rust-lang.org/stable/rust-by-example/custom_types/enum/c_like.html

Reply:
How are these list determined, can we add ed25519 that will simplify bridge (because reduced gas and we can do a full verification, and challenge step / watchdog can be removed)

Reply:
I would assume it follows https://github.com/ethereum/go-ethereum/blob/master/core/vm/contracts.go